src/VuData.cpp

```diff
@@ -42,31 +42,25 @@ using namespace vuplus;
 using namespace ADDON;
 using namespace P8PLATFORM;
 
-Vu::Vu() 
+Vu::Vu(const Settings &settings) 
+  : m_settings(settings)
 {
-  m_bIsConnected = false;
-  m_strServerName = "Vu";
   std::string strURL = "";
+  m_entryExtractor = std::unique_ptr<EpgEntryExtractor>(new EpgEntryExtractor(m_settings));
```

Comment:
Please put the member init into the init list, not in the body. Otherwise, the compiler generated init list code will first default construct `m_entryExtractor` and then, in the body the previously constructed instance will be assigned the value what you really want. This is inefficient. Always prefer init list over member init in ctor body.